tests/cloner_test.go

```diff
@@ -531,6 +531,73 @@ var _ = Describe("all clone tests", func() {
 			})
 			ClonerBehavior(cloneStorageClassName, "csivolumeclone")
 		})
+
+		Context("CloneStrategy on storageclass annotation", func() {
+			cloneType := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyCsiClone)
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				//TODO:  get storageClass `cloneStorageClassName` and annotate with `cloneType`
+				storageProfile, err = f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
```

Comment:
Shouldn't we be setting the annotation on the storage class here, instead of the storage profile. Something like this:
```go
  storageclass, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
  Expect(err).ToNot(HaveOccurred())
  if storageClass.GetAnnotations() == nil {
    storageClass.SetAnnotations(make(map[string]string))
  }
  storageClass.Annotations["cdi.kubevirt.io/clone-strategy"] = string(cloneType)
  _, err = f.K8sClient.StorageV1().StorageClasses().Update(context.TODO(), storageClass, metav1.UpdateOptions{})
  Expect(err).ToNot(HaveOccurred())
  Eventually(func() string {
    storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
    Expect(err).ToNot(HaveOccurred())
    return storageProfile.Status.CloneStrategy
  }, time.Minute, time.Second).Should(Equal(cloneType))
```
This updates the storage class, then poll the storage profile for a minute every second to check if the clone strategy matches what we set on the storage class.

Reply:
yep this makes sense, done